src/main/java/jenkinsci/plugins/influxdb/generators/SonarQubePointGenerator.java

```diff
@@ -181,22 +244,34 @@ protected String getResult(String url) throws IOException {
         }
     }
 
-    private String getSonarProjectURLFromBuildLogs(Run<?, ?> build) throws IOException {
+    private String[] getSonarProjectURLFromBuildLogs(Run<?, ?> build) throws IOException {
         String url = null;
+        String taskid = null;
+
         try (BufferedReader br = new BufferedReader(build.getLogReader())) {
             String line;
-            Pattern p = Pattern.compile(URL_PATTERN_IN_LOGS);
+            Pattern p_url = Pattern.compile(URL_PATTERN_IN_LOGS);
+            Pattern p_task = Pattern.compile(TASK_ID_PATTERN_IN_LOGS);
             while ((line = br.readLine()) != null) {
-                Matcher match = p.matcher(line);
+                Matcher match = p_url.matcher(line);
                 if (match.matches()) {
                     url = match.group(1);
+                } else {
+                    match = p_task.matcher(line);
+                    if (match.matches()) {
+                        taskid = match.group(1);
+                        break; // No need to search for other lines
+                    } 
                 }
             }
         }
-        return url;
+
+        String[] rtr_str_array = {url, taskid}; //return string array
+
+        return rtr_str_array;
     }
 
-    String getSonarProjectName(String url) throws URISyntaxException {
+    public String getSonarProjectName(String url) throws URISyntaxException {
```

Comment:
Why does this need to be public?

Reply:
My fault. Should be `private`. Thanks for the hint!